sdk/azcore/credential.go

```diff
@@ -16,6 +16,9 @@ type AuthenticationPolicyOptions struct {
 	// the list of OAuth2 authentication scopes used when requesting a token.
 	// This field is ignored for other forms of authentication (e.g. shared key).
 	Options TokenRequestOptions
+	// TokenProcessor is used to provide a custom implementation for checking the expires_on
+	// time of the token, when tokens should be refreshed, and setting custom headers.
+	TokenProcessor BearerTokenProcessor
```

Comment:
Can this be removed from here and passed directly to `NewBearerTokenPolicy()` instead?

Reply:
Yeah I was torn about where to put it. I'm fine with moving it onto the ctor

Reply:
Done...let me know how it's looking. The `defaultTokenProcessor` is what would exist in azidentity.